sql/dd/dd_table.cc

```diff
@@ -2814,17 +2816,19 @@ Encrypt_result is_tablespace_encrypted(THD *thd, const Table &t,
   auto unique_end = std::unique(tspids.begin(), valid_end);
 
   bool error = false;
+  String_type encryption_option;
   bool encrypted =
       std::any_of(tspids.begin(), unique_end, [&](const Object_id id) {
         bool is_encrypted = false;
         if (is_general_tablespace_and_encrypted(id, thd, &is_encrypted,
-                                                is_general_tablespace)) {
+                                                is_general_tablespace,
+                                                encryption_option)) {
           error = true;
           return false;
```

Comment:
How did the casting work here? Shouldn't it be like above code you wrote?

return {false, {false, ""}}; ?

Do we need the bool? can we infer from the string_type returned?